src/ebml/decoder.js

```diff
@@ -0,0 +1,261 @@
+            this.cursor,
+            this.cursor + tag.length,
+        );
+        const tagNum = Number.parseInt(tagStr, 16);
+        this.cursor += tag.length;
+        this.total += tag.length;
+        this.state = STATE_SIZE;
+
+        const tagObj = {
+            tag: tag.value,
+            tagStr,
+            type: EbmlDecoder.getSchemaInfo(tagNum).type,
+            name: EbmlDecoder.getSchemaInfo(tagNum).name,
+            start,
+            end: start + tag.length,
+        };
+
+        this.tagStack.push(tagObj);
+        debug(`read tag: ${tagStr}`);
+
+        return true;
+    }
+
+    readSize() {
+        const tagObj = this.tagStack[this.tagStack.length - 1];
+
+        debug(`parsing size for tag: ${tagObj.tagStr}`);
+
+        if (this.cursor >= this.buffer.length) {
+            debug('waiting for more data');
+
+            return false;
+        }
+
+        const size = tools.readVint(this.buffer, this.cursor);
+
+        if (size == null) {
+            debug('waiting for more data');
+
+            return false;
+        }
+
+        this.cursor += size.length;
+        this.total += size.length;
+        this.state = STATE_CONTENT;
+        tagObj.dataSize = size.value;
+
+        // unknown size
+        if (size.value === -1) {
+            tagObj.end = -1;
+        } else {
+            tagObj.end += size.value + size.length;
+        }
+
+        debug(`read size: ${size.value}`);
+
+        return true;
+    }
+
+    readContent() {
+        const { tagStr, type, dataSize, ...rest } = this.tagStack[
+            this.tagStack.length - 1
+        ];
+
+        debug(`parsing content for tag: ${tagStr}`);
+
+        if (type === 'm') {
+            debug('content should be tags');
+            this.push(['start', { tagStr, type, dataSize, ...rest }]);
+            this.state = STATE_TAG;
+
+            return true;
+        }
+
+        if (this.buffer.length < this.cursor + dataSize) {
+            debug(`got: ${this.buffer.length}`);
+            debug(`need: ${this.cursor + dataSize}`);
+            debug('waiting for more data');
+
+            return false;
+        }
+
+        const data = this.buffer.subarray(this.cursor, this.cursor + dataSize);
+        this.total += dataSize;
+        this.state = STATE_TAG;
+        this.buffer = this.buffer.subarray(this.cursor + dataSize);
+        this.cursor = 0;
+
+        this.tagStack.pop(); // remove the object from the stack
+
+        this.push([
+            'tag',
+            tools.readDataFromTag(
+                { tagStr, type, dataSize, ...rest },
+                Buffer.from(data),
+            ),
+        ]);
+
+        while (this.tagStack.length > 0) {
+            const topEle = this.tagStack[this.tagStack.length - 1];
+            if (this.total < topEle.end) {
+                break;
+            }
+            this.push(['end', topEle]);
+            this.tagStack.pop();
+        }
+
+        debug(`read data: ${data.toString('hex')}`);
```

Comment:
You should add the `debug.enabled` check as done in #37 here, too. It is a gigantic boost in performance.